cmd/archeio/main_test.go

```diff
@@ -76,30 +89,105 @@ func TestIntegrationMain(t *testing.T) {
 		t.Fatal("timed out waiting for archeio to be ready")
 	}
 
-	// TODO: fake being on AWS
-	testPull := func(image string) {
-		// nolint:gosec // this is not user suplied input ...
-		cmd := exec.Command("crane", "pull", testAddr+"/"+image, os.DevNull)
-		//cmd.Stderr = os.Stderr
-		out, err := cmd.CombinedOutput()
-		if err != nil {
-			t.Errorf("pull for %q failed: %v", image, err)
-			t.Error("output: ")
-			t.Error(string(out))
-			t.Fail()
-		}
+	// perform many test pulls ...
+	testCases := makeTestCases()
+	for i := range testCases {
+		tc := testCases[i]
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+			ref := testAddr + "/" + tc.Image
+			// ensure we supply fake IP info from test case
+			craneOpts := []crane.Option{crane.WithTransport(newFakeIPTransport(tc.FakeIP))}
+			// test fetching digest first
+			digest, err := crane.Digest(ref, craneOpts...)
+			if err != nil {
+				t.Errorf("Fetch digest for %q failed: %v", ref, err)
+			}
+			if digest != tc.Digest {
+				t.Errorf("Wrong digest for %q", ref)
+				t.Errorf("Received: %q", digest)
+				t.Errorf("Expected: %q", tc.Digest)
+			}
+			err = pull(ref, craneOpts...)
+			if err != nil {
+				t.Errorf("Pull for %q failed: %v", ref, err)
+			}
+		})
 	}
+}
 
-	// test pulling pause image
-	// TODO: test pulling more things
-	testPull("pause:3.1")
+func makeTestCases() []integrationTestCase {
+	wellKnownImages := []struct {
+		Name   string
+		Digest string
+	}{
+		{
+			Name:   "pause:3.1",
+			Digest: "sha256:f78411e19d84a252e53bff71a4407a5686c46983a2c2eeed83929b888179acea",
+		},
+		{
+			Name:   "pause:3.9",
+			Digest: "sha256:7031c1b283388d2c2e09b57badb803c05ebed362dc88d84b480cc47f72a21097",
+		},
+	}
+
+	interestingIPs := []struct {
+		Name string
+		IP   string
+	}{
+		{
+			Name: "GCP",
+			IP:   "35.220.26.1",
```

Comment:
TODO: very low priority: ensure these remain valid IPs (e.g. we could cross check with our own package for this)

unlikely to change + checking against the same code we'll use in the server is only slightly better than as-is